speedy-recovery-client/src/components/secured/core/SecuredMainPage.js

```diff
@@ -81,14 +81,20 @@ class SecuredMainPage extends Component {
   }
 
   componentWillMount() {
-    this.updateStateAppointments();
+    clearTimeout(this.updateStateAppointments);
   }
 
   updateStateAppointments = () => {
-    // TODO: Query SMART, display appointment data for current user
-    const appointments = fhirExampleApp.map((app) => fhirMapAppointment(app, this.props.fhirVersion));
+    FhirDataQueryingService.getUserAppointment(this.props.user.id)
+      .then(appointment => {
+        const appointments = this.handleAppointment(appointment);
+        this.setState({ appointments });
+      }).catch(errorMessage=>{this.setState([])});
+  };
 
-    this.setState({ appointments });
+  handleAppointment = appointment => {
+    const appointments = appointment.data.entry.map((app) => (fhirMapAppointment(app.resource, this.props.fhirVersion)));
```

Comment:
Sorry to nitpick but you could just return this line?

Reply:
yep, good suggestion!